src/utils/filesystem_path.cpp

```diff
@@ -141,7 +141,11 @@ class FilesystemPathImplWindows : public FilesystemPathImpl
 
 	virtual bool isFile() override
 	{
-		return !isDirectory();
+		WIN32_FIND_DATA ffd;
+		if (FindFirstFile(_path.c_str(), &ffd) == reinterpret_cast<HANDLE>(-1))
```

Comment:
Couldn't we replace `== reinterpret_cast<HANDLE>(-1)` with the more readable `== INVALID_HANDLE_VALUE`? I have taken it from [this example](https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-findfirstfilea#examples).

Reply:
Yes. I have gone ahead and applied this to isDirectory as well.